src/main/java/com/github/markusbernhardt/seleniumlibrary/locators/ElementFinder.java

```diff
@@ -301,33 +289,43 @@ protected static Strategy parseLocator(FindByCoordinates findByCoordinates, Stri
 		return strategy;
 	}
 
-	protected static void parseTag(FindByCoordinates findByCoordinates, Strategy strategy, String tag) {
+	protected static void parseTag(FindByCoordinates findByCoordinates, String tag) {
 		if (tag == null) {
 			return;
 		}
+
 		tag = tag.toLowerCase();
-		Map<String, String> constraints = new TreeMap<String, String>();
-		if (tag.equals("link")) {
-			tag = "a";
-		} else if (tag.equals("image")) {
-			tag = "img";
-		} else if (tag.equals("list")) {
-			tag = "select";
-		} else if (tag.equals("text area")) {
-			tag = "textarea";
-		} else if (tag.equals("radio button")) {
-			tag = "input";
-			constraints.put("type", "radio");
-		} else if (tag.equals("checkbox")) {
-			tag = "input";
-			constraints.put("type", "checkbox");
-		} else if (tag.equals("text field")) {
-			tag = "input";
-			constraints.put("type", "text");
-		} else if (tag.equals("file upload")) {
-			tag = "input";
-			constraints.put("type", "file");
-		}
+		Map<String, String> constraints = new TreeMap<>();
+        switch (tag) {
```

Comment:
There is something wrong with indentation.

Reply:
Done